### ISSUE TITLE
Setting LogFilePath config in etcd does not stop felix.log being written to.

### ISSUE THREAD

**Issue by robbrockbank** [projectcalico/felix#803]
In calico-docker, when creating a calico-node, we ensure a number of globals are configured in etcd.  This includes setting the LogFilePath to "none" so that we only get logs written to screen (we use our own log capture/rotation).

A value of "none" does not seem to stop logging to the default /var/log/calico/felix.log file.

If you set LogFilePath=None in the felix config file, there is no logging to this file as expected - just when setting in etcd.

**Comment by wrouesnel** [comment]
So I looked into this and hopefully will submit a patch soon: but basically this is a product of how the configuration is read - it's called twice, once for the environment and once for the config file. The problem is the logging parameters have default values which aren't set in the environment, but logging is setup anyway:

```
        self.add_parameter("LogFilePath",
                           "Path to log file", "/var/log/calico/felix.log")
        self.add_parameter("EtcdDriverLogFilePath",
                           "Path to log file for etcd driver",
                           "/var/log/calico/felix-etcd.log")
```

The result is Felix always tries to create/write to /var/log/calico a little bit. I'm thinking these should default to None?

**Comment by fasaxc** [comment]
We need to be able to log before we connect to etcd in order to diagnose connection problems.  @robbrockbank should we be enhancing calico/node to allow for env var overrides and/or felix.cfg settings instead?

**Comment by hectorj2f** [comment]
@fasaxc I'd be quite interested on your proposal https://github.com/projectcalico/calico/issues/803#issuecomment-245559675

**Comment by fasaxc** [comment]
Partially addressed by #1157

**Comment by fasaxc** [comment]
I think this is fixed under #1202
